modin/core/dataframe/pandas/interchange/dataframe_protocol/column.py

```diff
@@ -187,18 +187,21 @@ def _dtype_from_primitive_pandas_dtype(
         )
 
     @property
-    def describe_categorical(self) -> Dict[str, Any]:
-        if not self.dtype[0] == DTypeKind.CATEGORICAL:
-            raise RuntimeError(
+    def describe_categorical(self) -> CategoricalDescription:
+        if self.dtype[0] != DTypeKind.CATEGORICAL:
+            raise TypeError(
                 "`describe_categorical only works on a column with "
                 + "categorical dtype!"
             )
 
         pandas_series = self._col.to_pandas().squeeze(axis=1)
+        cat_frame = self._col.from_pandas(
```

Comment:
Not sure this is semantically correct because we call `from_pandas` from already created Modin DF to create other Modin DF. What about `modin.pandas.utils.from_pandas`?

Reply:
`modin.pandas.utils.from_pandas` creates a high-level `modin.pandas.DataFrame` when here we need a low-level `modin.core.dataframe...`. This code uses `self._col` instead of `PandasOn*Dataframe.from_pandas` because we don't know the actual class for the `PandasOn*Dataframe` before execution, the class depends on `self._col` and is decided at the runtime.

Reply:
> modin.pandas.utils.from_pandas creates a high-level modin.pandas.DataFrame when here we need a low-level modin.core.dataframe.... 

I understand we use here a low-level dataframe. By using `df = modin.pandas.utils.from_pandas`, we can get the modin frame as `df._query_compiler._modin_frame`.

> This code uses self._col instead of PandasOn*Dataframe.from_pandas because we don't know the actual class for the PandasOn*Dataframe before execution, the class depends on self._col and is decided at the runtime.

Yes, that is a known issue regarding the use of `modin.pandas.utils.from_pandas`. However, Column is not an object that the user tries to create first. Actually, the column is not supposed to be an object the user interacts with. Also, I still feel the use of `self._col.from_pandas` to create other df a bit confusing. Probably, if someone feels the use of `modin.pandas.utils.from_pandas` is not relevant here, we can use `type(self._col).from_pandas` instead.

Reply:
what's the difference between `self._col.from_pandas` and `type(self._col).from_pandas`, if `.from_pandas` is a classmethod itself?.. what's the point?

Reply:
`self._col.from_pandas` to create other df looks a bit confusing to me. `type(self._col).from_pandas` feels more natural to create an object as we see that we create other df from a class method, not from an instance.

Reply:
agreed, fixed